src/main/java/dk/cngroup/lentils/service/StatusService.java

```diff
@@ -14,15 +14,35 @@
 public class StatusService {
 
     private StatusRepository statusRepository;
+    private CypherService cypherService;
+
+    @Autowired
+    public void setCypherService(final CypherService cypherService) {
+        this.cypherService = cypherService;
+    }
 
     @Autowired
     public StatusService(final StatusRepository statusRepository) {
         this.statusRepository = statusRepository;
     }
 
     public void markCypher(final Cypher cypher, final Team team, final CypherStatus cypherStatus) {
-        Status status = getStatusByTeamAndCypher(team, cypher);
-        saveNewStatus(status, cypherStatus);
+            Status status = getStatusByTeamAndCypher(team, cypher);
+            saveNewStatus(status, cypherStatus);
+    }
+
+    public void solveCypher(final Cypher cypher, final Team team) {
+        markCypher(cypher, team, CypherStatus.SOLVED);
+        if (cypherService.getNext(cypher.getStage()) != null) {
+            markCypher(cypherService.getNext(cypher.getStage()), team, CypherStatus.PENDING);
+        }
+    }
+
+    public void skipCypher(final Cypher cypher, final Team team) {
+        markCypher(cypher, team, CypherStatus.SKIPPED);
+        if (cypherService.getNext(cypher.getStage()) != null) {
```

Comment:
Tady je ted duplikace.
Dotaz. Jak je to, kdyz organizator vraci sifru do stavu in progress nebo tak neco?

Jestli tady nevzrobit nejakou tabulku, kde prvni pole je pozadovany status, druhe je status, ktery se ma nastavit pro dalsi sifru a neudelat jednu generickou metodu pro zmenu stavu.

Reply:
Máš pravdu, s tím vrácením stavu u organizátora jsem nepočítal, zítra se na to vrhnu :)